swarm/storage/mru/topic.go

```diff
@@ -0,0 +1,56 @@
+package mru
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/ethereum/go-ethereum/common/bitutil"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/swarm/storage"
+)
+
+const topicLength = storage.KeyLength
+
+type Topic struct {
+	content [topicLength]byte
+}
+
+func NewTopic(name string, relatedContent storage.Address) Topic {
+	var topic Topic
+	copy(topic.content[:], relatedContent)
+	nameLength := len(name)
+	if nameLength > topicLength {
+		nameLength = topicLength
+	}
+	bitutil.XORBytes(topic.content[:], topic.content[:], []byte(name)[:nameLength])
+	return topic
+}
+
+func (t *Topic) Hex() string {
```

Comment:
`ToHex` (since we have `FromHex`)

Reply:
I just looked at the `common` package and saw that `common.Address` has an `Hex()` method with that name, so I think we should leave it as is.

Reply:
I see, it's not consistent in `common` either. We have two `Hex()` and one `ToHex()` there. Good thing the majority is always right.

http://quotespictures.com/wp-content/uploads/2015/03/famous-celebrity-quote-by-henrik-ibsen-the-majority-is-always-wrong-the-minority-is-rarely-right.jpg